geopandas/tests/test_plotting.py

```diff
@@ -628,6 +798,43 @@ def test_style_kwargs(self):
         ax = self.df.plot(markersize=10)
         assert ax.collections[2].get_sizes() == [10]
 
+        # linestyle
+        #   single
+        for ax in [
+            self.series.plot(linestyle=":", linewidth=1),
+            self.df.plot(linestyle=":", linewidth=1),
+        ]:
+            np.testing.assert_array_equal(
+                [(0.0, [1.0, 1.65])], ax.collections[0].get_linestyle()
+            )
+        #   multiple
+        # ls = ["solid", "dotted", "dashdot"]
+        # exp_ls = [_style_to_linestring_onoffseq(l, 1) for l in ls]
+        # for ax in [self.series.plot(linestyle=ls, linewidth=1),
+        #            self.series.plot(linestyles=ls, linewidth=1),
+        #            self.df.plot(linestyles=ls, linewidth=1)]:
+        #     np.testing.assert_array_equal(exp_ls, ax.collections[0].get_linestyle())
+
+        # linewidth
+        #   single
+        ax = self.df.plot(linewidth=2)
+        np.testing.assert_array_equal([2], ax.collections[0].get_linewidths())
+        #   multiple
+        for ax in [
+            self.series.plot(linewidths=[2, 4, 5.5]),
+            self.series.plot(linewidths=[2, 4, 5.5]),
+            self.df.plot(linewidths=[2, 4, 5.5]),
+        ]:
+            np.testing.assert_array_equal(
+                [2, 4, 5.5], ax.collections[0].get_linewidths()
```

Comment:
This test fails, because the non uniform series is broken down into uniform bits, which are then plotted individually, each with the entire kwargs dictionary as parameter. (`plotting.py`, lines 390-420) Therefore, there are actually 3 elements in `ax.collections`, the polygons, the lines, the points (in that order). 

Changing the series plotting function to correctly handle the passing of a listlike kwargs parameter and a non-uniform series is not trivial. The caller will expect the elements of the keyword values list to line up one-to-one with the shapes in the series, which would require some checking and processing of the kwargs parameters (i.e., check if they are listlike, and if so, split them into a part for the polygons, a part for the lines, and a part for the points)

Anyway, my suggestion is to drop the tests with listlike keyword values from the plot series tests, and start a discussion as to if and how this should be handled, and allow only single valued kwargs for non uniform series.

Reply:
Yes, this is the issue which needs to be resolved as well. It is already reported in #1379 as it does affect current handling of `color` as well. We can skip that for now with a reference to that issue and figure out how to fix that later, as another PR.